soco/snapshot.py

```diff
@@ -90,7 +97,12 @@ def snapshot(self):
         if self.media_uri.split(':')[0] != 'x-rincon':
             self.is_coordinator = True
         if self.media_uri.split(':')[0] == 'x-rincon-queue':
-            self.is_playing_queue = True
+            if self.media_uri.split('#')[1] == '0':
```

Comment:
Do you have some idea how robust this check is. I can't really tell from the code. Can you comment, maybe just here, what the uris might look like in the different cases

Reply:
I think this is robust from analysis and current use. The is_playing_queue has been in use in Snapshot since it's inception. The addition here is to differentiate the types of queues now Sonos have introduced a cloud queue to work with Alexa.

Special URI's in Sonos:

x-sonosapi-stream: ....                   radio stream
x-sonosapi-radio: .....                     radio stream / Amazon Station / Google mixes

x-rincon-queue:RINCON_000E5859E49601400#0   - local queue (note: #0 always zero for local queue)

x-rincon-queue:RINCON_000E5859E49601400#6   - cloud queue (note: #x where x changes with each queue)

x-rincon:RINCON_000E5859E49601400   -  slave pointing to coordinator 

Although I have never seen a slave not start 'x-rincon:' the snapshot code has been changed to use device.is_coordinator in place of this #529 as there were reported issues #519 which on analysis appeared to show what snapshot thought was not a slave was a slave in SoCo!  As intermittent hard to debug, but this approach makes it consistent.

Hope this makes sense - may be worth adding this in a documentation area so we can build on it as we discover more....
Shout if you want more.

Cheers David

Reply:
Ah makes sense and I agree should be robust. In general, for code that isn't immediately obvious, like string determination with split and indexing, I think it is always worth it to explain in a comment above with an example. So basically a small comment around your:

```text
x-rincon-queue:RINCON_000E5859E49601400#0 - local queue (note: #0 always zero for local queue)

x-rincon-queue:RINCON_000E5859E49601400#6 - cloud queue (note: #x where x changes with each queue)

x-rincon:RINCON_000E5859E49601400 - slave pointing to coordinator
```
observation would be great.